pkg/domains/otellog/otel_log.go

```diff
@@ -0,0 +1,247 @@
+}
+
+type domain struct{ *impl.Domain }
+
+func (d *domain) Query(s string) (korrel8r.Query, error) {
+	c, s, err := impl.ParseQuery(d, s)
+	if err != nil {
+		return nil, err
+	}
+	return NewQuery(c.(Class), s), nil
+}
+
+const (
+	StoreKeyLoki      = "loki"
+	StoreKeyLokiStack = "lokiStack"
+)
+
+func (*domain) Store(s any) (korrel8r.Store, error) {
+	cs, err := impl.TypeAssert[config.Store](s)
+	if err != nil {
+		return nil, err
+	}
+	hc, err := k8s.NewHTTPClient(cs)
+	if err != nil {
+		return nil, err
+	}
+
+	loki, lokiStack := cs[StoreKeyLoki], cs[StoreKeyLokiStack]
+	switch {
+
+	case loki != "" && lokiStack != "":
+		return nil, fmt.Errorf("can't set both loki and lokiStack URLs")
+
+	case loki != "":
+		u, err := url.Parse(loki)
+		if err != nil {
+			return nil, err
+		}
+		return NewPlainLokiStore(u, hc)
+
+	case lokiStack != "":
+		u, err := url.Parse(lokiStack)
+		if err != nil {
+			return nil, err
+		}
+		return NewLokiStackStore(u, hc)
+
+	default:
+		return nil, fmt.Errorf("must set one of loki or lokiStack URLs")
+	}
+}
+
+// Class is the log_type name.
+type Class string
+
+func (c Class) Domain() korrel8r.Domain                     { return Domain }
+func (c Class) Name() string                                { return string(c) }
+func (c Class) String() string                              { return impl.ClassString(c) }
+//func (c Class) Preview(o korrel8r.Object) (line string)     { return Preview(o) }
+
+// Preview extracts the message from a OTEL log record.
+/* func Preview(x korrel8r.Object) (line string) {
+	if m := x.(Object)["Body"]; m != nil {
+		s, _ := m.(string)
+		return s
+	}
+	return ""
+} */
+
+func (c Class) Unmarshal(b []byte) (korrel8r.Object, error) {
+	return impl.UnmarshalAs[Object](b)
+}
+
+func (c Class) ID(o korrel8r.Object) any {
+	if log, _ := o.(*OtelLog); log != nil {
+		return log.Timestamp
+	}
+	return nil
+}
+
+// Object represents an OpenTelemetry [log]
+//
+// [log]: https://opentelemetry.io/docs/concepts/signals/logs/#log-record
+type Object = *OtelLog
+
+// OtelLog is an OpenTelemetry [log], the smallest unit of work for logging.
+//
+// Implements the OpenTelemetry API [Spec].
+//
+// Otellog: [https://opentelemetry.io/docs/concepts/signals/logs/#log-record]
+type OtelLog struct {
+	Body        string                 `json:"body"`           // ← Log line
+	Severity    string                 `json:"severityText"`   // ← Can become a label
+	Timestamp   time.Time                 `json:"timestamp"`      // ← Loki timestamp
+	Attributes  map[string]any `json:"attributes,omitempty"` // ← Becomes atreams and structured metadata
+}
+
+// Query is a LogQL query string
+type Query struct {
+	logQL string // `json:",omitempty"`
+	class Class  // `json:",omitempty"`
+}
+
+func NewQuery(c Class, logQL string) korrel8r.Query {
+	logQL = strings.TrimSpace(logQL)
+	if c == "" {
+		c = logQueryClass(logQL)
+	}
+	return Query{class: c, logQL: logQL}
+}
+
+const (
```

Comment:
Do these classes still make sense,  or should we treat all otel logs as instances of a single class?